core/src/main/scala/kafka/log/UnifiedLog.scala

```diff
@@ -2128,7 +2182,7 @@ object LogMetricNames {
 
 case class RetentionMsBreach(log: UnifiedLog) extends SegmentDeletionReason {
   override def logReason(toDelete: List[LogSegment]): Unit = {
-    val retentionMs = log.config.retentionMs
+    val retentionMs = log.config.remoteLogConfig.localRetentionMs
```

Comment:
retention time depends on remote storage being enabled, right? Ditto in line 2284.